build.assets/images.mk

```diff
@@ -2,7 +2,8 @@
 # by other Makefiles
 # These values may need to be updated in `dronegen/container_image_products.go` if
 # they change here
-BUILDBOX_VERSION ?= teleport12
+# TODO(for the person updating to v13): drop the minor version part
+BUILDBOX_VERSION ?= teleport12.1
```

Comment:
This will break the Drone as the image update also requires updating the `.drone.yml` (check the comment above).
We've also never published `.1` release on master. Any changes to solve this issue without changing the image tag?

Reply:
@r0mant also challenged me on the image tag. Changing the binary in the current buildbox tag will cause all pipelines from the past 3 weeks to be non-retyable and all commits on master between the v11 -> v12 change and this commit would have `make test` failing.

For the drone part I wanted to build the image in this PR but start using it in the next one. If that's not possible I'll bundle all changes in a single PR and retry all jobs until the `teleport12.1` image is built

Reply:
Yeah, I think updating buildbox version here is not too bad. We gotta have some way to introduce backwards incompatible changes to the buildboxes and using version for that seems ok. Just need to make sure to update all pipelines to use this version after it's built.

Reply:
Based on our discussions I changed the approach and installed the plugin a second time in a different location. The plugin lookup path is controlled through the `HELM_PLUGINS` variable. This variable will be set in:
- `build.assets/Makefile`
- `.cloudbuild/ci/unit-tests.yaml`
- `.github/workflows/unit-tests-helm.yaml`

@jakule @r0mant could you both re-review the PR and confirm this approach is saner?